Packs/SolarWinds/Integrations/SolarWinds/SolarWinds.py

```diff
@@ -0,0 +1,657 @@
+
+def validate_fetch_incidents_parameters(params: dict) -> dict:
+    """
+    Validate fetch incidents params, throw ValueError on non-compliant  arguments
+
+    :param params: dictionary of parameters to be tested for fetch_incidents
+
+    :rtype: ``dict``
+    return: dictionary containing valid parameters
+    """
+
+    max_fetch = arg_to_number(params.get("max_fetch", DEFAULT_MAX_FETCH))
+    if (max_fetch is None) or (not 0 < max_fetch <= 1000):
+        raise ValueError(ERR_MSG["INVALID_MAX_FETCH"])
+    params["max_fetch"] = max_fetch
+
+    first_fetch = params.get("first_fetch")
+    first_fetch_time = arg_to_datetime(first_fetch, required=True,
+                                       settings={'TIMEZONE': "UTC", "RETURN_AS_TIMEZONE_AWARE": True})
+    if first_fetch_time is None:
+        raise ValueError(ERR_MSG["INVALID_FIRST_FETCH"])
+    params["first_fetch"] = first_fetch_time
+
+    severities = params.get("severities", [])
+    for severity in severities:
+        if severity.upper() not in SEVERITIES_MAP:
+            raise ValueError(ERR_MSG["SEVERITIES_ERROR"].format(SEVERITIES_MAP))
+    params["severities"] = severities
+
+    return params
+
+
+def prepare_query_for_fetch_alerts(last_run: dict, params: dict) -> str:
+    """
+    Return the query to fetch alerts
+
+    :param last_run: last run returned by function demisto.getLastRun.
+
+    :param params: arguments for fetch-incident.
+
+    :rtype: ``str``
+    :return: query string to send in request
+    """
+    max_fetch = params.get("max_fetch")
+    first_fetch = params.get("first_fetch")
+    severities = params.get("severities")
+    object_types = params.get("object_types")
+
+    query = QUERY_PARAM["FETCH_ALERTS"].format(max_fetch)
+
+    # The AlertActiveIDs are unique for alerts, so it is used instead of timestamp for the last run for simplicity
+    if last_run.get('alert_active_id'):
+        query += f" WHERE A.AlertActiveID>{last_run.get('alert_active_id')}"
+    else:
+        query += f" WHERE A.TriggeredDateTime>='{first_fetch.strftime(DATE_FORMAT)}'"  # type: ignore
+
+    if severities:
+        query += " AND (" + " OR ".join(
+            [f"C.Severity={SEVERITIES_MAP.index(severity)}" for severity in severities]) + ")"
+    if object_types:
+        query += " AND (" + " OR ".join(
+            [f"C.ObjectType='{object_type}'" for object_type in object_types]) + ")"
+
+    query += " ORDER BY A.AlertActiveID"
+
+    return query
+
+
+def prepare_query_for_fetch_events(last_run: dict, params: dict) -> str:
+    """
+    Returns the query to fetch events
+
+    :param last_run: last run returned by function demisto.getLastRun.
+
+    :param params: arguments for fetch-incident.
+
+    :rtype: ``str``
+    :return: query string to send in request
+    """
+    max_fetch = params.get("max_fetch")
+    first_fetch = params.get("first_fetch")
+    event_types = params.get("event_types")
+
+    query = QUERY_PARAM["FETCH_EVENTS"].format(max_fetch)
+
```

Comment:
```suggestion
# The EventIDs are unique for events, so it is used instead of timestamp for the last run for simplicity
```
might be nice to add the same comment here.